akka-sample-sharding-scala/src/main/scala/sample/sharding/EdgesApp.scala

```diff
@@ -0,0 +1,155 @@
+import scala.util.{ Failure, Random, Success }
+
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior, PostStop }
+import akka.actor.typed.scaladsl.{ Behaviors, TimerScheduler }
+import com.typesafe.config.ConfigFactory
+
+/** Simulate devices and stations.
+ * In the wild, each station would run its own simple system,
+ * taking samples and posting via HTTPS.
+ */
+object EdgesApp {
+
+  val WeatherStations = 50
+  // temperature, pressure, precipitation, wind speed...
+  val DevicesPerStation = 1
+
+  val random = new Random()
+
+  def main(args: Array[String]): Unit = {
+    val config =
+      ConfigFactory
+        .parseString(s"akka.remote.artery.canonical.port = 0")
+        .withFallback(ConfigFactory.load("weather-station.conf"))
+
+    // simple sample, using one node on local machine
+    val rootBehavior = Behaviors.setup[Nothing] { context =>
+      (0 to WeatherStations).foreach { sid =>
+        context.spawn(Station(sid, DevicesPerStation, 8080), s"station-$sid")
+      }
+
+      Behaviors.empty
+    }
+
+    ActorSystem[Nothing](rootBehavior, "SimulatedDevices", config)
+  }
+}
+
+object Station {
+
+  def apply(stationId: Int, devices: Int, port: Int): Behavior[Device.Data] =
+    Behaviors.setup { context =>
+      context.log.info("Started station with {} devices.", devices)
+      val http = context.spawn[Device.Data](Post(stationId, port), s"http-post-$stationId")
+
+      (0 to devices).foreach { deviceId =>
+        context.spawn(Device(stationId, deviceId, http, 5.second, 10), s"device-$deviceId")
+      }
+
+      Behaviors.empty
+    }
+}
+
+/**
+ * Any given weather station has a unique ID, and can sample temperature, wind speed, precipitation, etc.
+ * from multiple devices.
+ *
+ * For a simple example, this does one type of reading.
+ */
+object Post {
+
+  def apply(stationId: Int, port: Int): Behavior[Device.Data] =
+    Behaviors.setup[Device.Data] { context =>
+      import akka.http.scaladsl.Http
+      import akka.http.scaladsl.model.{ ContentTypes, HttpEntity, HttpMethods, HttpRequest }
+      import context.executionContext
+
+      import akka.actor.typed.scaladsl.adapter._
+      val http = Http(context.system.toClassic)
+
+      Behaviors.receiveMessage[Device.Data] {
+        case Device.Data(sid, did, data) =>
+          // TODO
+          val values = data.mkString(",")
+          val json = s"""{"stationId":$sid,"deviceId":$did,"data":[$values]}"""
+          context.log.info("Read: {}", json)
+
+          // See https://doc.akka.io/docs/akka-http/current/common/index.html
+          // for marshalling, unmarshalling, json support in the wild.
+          val httpEntity = HttpEntity(ContentTypes.`application/json`, json)
+
+          http
+            .singleRequest(
+              HttpRequest(method = HttpMethods.POST, uri = s"http://localhost:$port/temperature", entity = httpEntity))
+            .onComplete {
+              case Success(res) => context.log.info(res.toString())
```

Comment:
I think you need to consume the response body here or else it will backpressure not allow any more requests on the same connection https://doc.akka.io/docs/akka-http/current/implications-of-streaming-http-entity.html#consuming-the-http-response-entity-client-

Reply:
Thanks, done.